nimbus-core/src/main/java/com/antheminc/oss/nimbus/domain/model/state/repo/DefaultModelRepositoryFactory.java

```diff
@@ -53,5 +52,13 @@ public ModelRepository get(Repo.Database db) {
 	public ModelPersistenceHandler getHandler(Repo repo) {
 		return getBeanResolver().get(ModelPersistenceHandler.class, repo.value().name()+"_handler");
 	}
+
+	@Override
+	public ModelRepository get(ModelConfig<?> mConfig) {
+		if(mConfig.isRemote()) {
```

Comment:
Pls change the lookup happening on each call to singleton post construct init or constructor. The Enum values are known (REMOTE, LOCAL), as such a map (Key=Enum.Scope, Value=ModelRepoImpl) could be passed in part of init/constructor. This get method can then just do a map lookup and not be concerned on how the beans are constructed.